source/dem/dem.cc

```diff
@@ -888,22 +1116,39 @@ DEMSolver<dim>::solve()
           checkpoint_step = false;
 
           // Execute broad search by filling containers of particle-particle
+          // contact pair candidates and containers of particle-wall
           // contact pair candidates
-          container_manager.execute_particle_particle_broad_search(
-            particle_handler, has_periodic_boundaries);
+          if (has_disabled_contacts && contact_build_number > 1)
```

Comment:
Your case is the exception, it is not the norm, so the order of the if-else should be reversed.